mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -22,11 +23,20 @@ class ElasticsearchStore(AbstractStore):
 
     def __init__(self, store_uri: str = None, artifact_uri: str = None) -> None:
         self.is_plugin = True
-        url = store_uri.split('/')[2]
-
-        connections.create_connection(hosts=[url])
+        connections.create_connection(hosts=[urllib.parse.urlparse(store_uri).netloc])
         super(ElasticsearchStore, self).__init__()
 
+    def hit_to_mlflow(self, model, **kwargs):
+        return model(**kwargs)
+
+    def list_experiments(self, view_type: str = ViewType.ACTIVE_ONLY) -> List[Experiment]:
+        stages = LifecycleStage.view_type_to_stages(view_type)
+        response = Search(index="mlflow-experiments").filter("terms",
+                                                             lifecycle_stage=stages).execute()
+        return [self.hit_to_mlflow(Experiment, experiment_id=e.meta.id, name=e.name,
```

Comment:
The issue here with your generical function is that you will have to type again all the keyword argument matching if you ever need to perform another conversion somewhere else in your code.

The proposal I had yesterday was more to define one specific conversion function for each mlflow entity, so that the mapping between the hit fields and the entity is written once.